fix: remove experimental spread init templates (#2118)
canonical/charmcraft: docs/reference/changelog.rst

  For a complete list of commits, see the `X.Y.Z`_ release on GitHub.

3.3.2 (2025-01027)
------------------

This release contains some bug fixes for the 3.3 branch.

Command line
============

> This release removes experimental test templates added to the project when
> running ``charmcraft init``. Existing projects will not be affected.

Reviewer: > This release removes... added to

could be potentially confusing. Here's a fix:

```suggestion
Stop the generation of experimental test templates during project initialisation.
Existing projects aren't affected.
```
